sunpy/timeseries/sources/goes.py

```diff
@@ -173,8 +174,8 @@ def _parse_hdus(cls, hdulist):
         else:
             raise ValueError("Don't know how to parse this file")
 
-        times = [start_time + datetime.timedelta(seconds=int(np.floor(s)),
-                                                    microseconds=int((s - np.floor(s)) * 1e6)) for s in seconds_from_start]
+        times = [start_time + TimeDelta(s*u.second) for s in seconds_from_start]
```

Comment:
Also I don't think you need to loop here, off the top of my head I think this should work:
```python
times = start_time + TimeDelta(seconds_from_start*u.s)
```

Reply:
That is so beautiful!! :heart_eyes_cat: